contrib/errorprone/src/python/pants/contrib/errorprone/tasks/errorprone.py

```diff
@@ -0,0 +1,140 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.jvm.subsystems.shader import Shader
+from pants.backend.jvm.targets.jar_dependency import JarDependency
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.util.dirutil import safe_mkdir
+from pants.util.strutil import safe_shlex_split
+from twitter.common.collections import OrderedSet
+
+
+class ErrorProne(NailgunTask):
+  """Check Java code for Error Prone violations.  See http://errorprone.info/ for more details."""
+
+  _ERRORPRONE_MAIN = 'com.google.errorprone.ErrorProneCompiler'
+  _JAVA_SOURCE_EXTENSION = '.java'
+
+  @classmethod
+  def register_options(cls, register):
+    super(ErrorProne, cls).register_options(register)
+
+    register('--skip', type=bool, help='Skip Error Prone.')
+    register('--transitive', default=False, type=bool,
```

Comment:
> We always pass the explicit list of targets we want to check when we use Error Prone so we could remove this option?

What the `invalidate_dependents` option refers to is whether the tool you are running (errorprone) uses the dependencies (classpath) of the target you're building. And it does, because you give it the transitive classpath. So making this an option does not make sense.